lib/sps/detail/Sps.cpp

```diff
@@ -0,0 +1,235 @@
+	// allocate memory gpu
+	allocate_memory_gpu(dedispersion_plan);
+	//
+	output_sps.resize(*output_size/sizeof(float));
+
+	printf("\nDe-dispersing...\n");
+
+	int t, dm_range;
+	float tsamp_original = _tsamp;
+	int maxshift_original = _maxshift;
+
+	//
+	float *out_tmp = NULL;
+	out_tmp = (float *) malloc(( _t_processed[0][0] + _maxshift ) * _max_ndms * sizeof(float));
+	memset(out_tmp, 0.0f, _t_processed[0][0] + _maxshift * _max_ndms * sizeof(float));
+
+	// count the number of times that analysis is called - used to fill the output buffer of analysis
+	int analysis_call = 0;
+	for (t = 0; t < _num_tchunks; ++t)
+	{
+
+		printf("\nt_processed:\t%d, %d", _t_processed[0][t], t);
+
+		load_data(-1, _in_bin, _d_input, &_input_buffer[(long int) ( inc * _nchans )],
+							_t_processed[0][t], _maxshift, _nchans, _dmshifts);
+
+		if (user_input.get_enable_zero_dm())
+			zero_dm(_d_input, _nchans, _t_processed[0][t]+_maxshift);
+
+		corner_turn(_d_input, _d_output, _nchans, _t_processed[0][t] + _maxshift);
+		int oldBin = 1;
+
+		for (dm_range = 0; dm_range < _range; ++dm_range)
+		{
+			printf("\n\n%f\t%f\t%f\t%d", _dm_low[dm_range], _dm_high[dm_range],	_dm_step[dm_range], _ndms[dm_range]), fflush(stdout);
+			printf("\nAmount of telescope time processed: %f", tstart_local);
+
+			_maxshift = maxshift_original / _in_bin[dm_range];
+
+			cudaDeviceSynchronize();
+			load_data(dm_range, _in_bin, _d_input, &_input_buffer[(long int) ( inc * _nchans )], _t_processed[dm_range][t], _maxshift, _nchans, _dmshifts);
+
+			if (_in_bin[dm_range] > oldBin)
+			{
+				bin_gpu(_d_input, _d_output, _nchans, _t_processed[dm_range - 1][t] + _maxshift * _in_bin[dm_range]);
+				( _tsamp ) = ( _tsamp ) * 2.0f;
+			}
+
+			dedisperse(dm_range, _t_processed[dm_range][t], _in_bin, _dmshifts, _d_input, _d_output, _nchans,
+			           ( _t_processed[dm_range][t] + _maxshift ), _maxshift, &_tsamp, _dm_low, _dm_high, _dm_step, _ndms);
+
+			_gpu_output_size = _ndms[dm_range] * ( _t_processed[dm_range][t] ) * sizeof(float);
+
+			save_data(_d_output, out_tmp, _gpu_output_size);
+
+			for (int k = 0; k < _ndms[dm_range]; ++k)
+			{
+				memcpy(&(*output_buffer)[dm_range][k][inc / _in_bin[dm_range]], &out_tmp[k * _t_processed[dm_range][t]],
+							sizeof(float) * _t_processed[dm_range][t]);
+			}
+
+			if (user_input.get_output_dmt() == 1)
+				write_output(dm_range, _t_processed[dm_range][t], _ndms[dm_range], gpu_memory, out_tmp, _gpu_output_size, _dm_low, _dm_high);
+
+			if (user_input.get_enable_analysis() == 1)
+			{
+				analysis(dm_range, tstart_local, _t_processed[dm_range][t],
+						( _t_processed[dm_range][t] + _maxshift ), _nchans,
+						_maxshift, _max_ndms, _ndms, _out_bin, _sigma_cutoff,
+						_d_output, _dm_low, _dm_high, _dm_step, _tsamp, output_sps, analysis_call);
+				analysis_call += 1;
+
+			}
+			oldBin = _in_bin[dm_range];
+		}
+
+
+		memset(out_tmp, 0.0f, _t_processed[0][0] + _maxshift * _max_ndms * sizeof(float));
```

Comment:
but 
out_tmp = (float *) malloc(( _t_processed[0][0] + _maxshift ) * _max_ndms * sizeof(float));

notice positioning of brackets! Which is right?